src/main/scala/wdlTools/eval/stdlib/BaseStdlib.scala

```diff
@@ -259,83 +298,98 @@ case class StdlibV1(opts: Options, evalCfg: EvalConfig, docSourceUrl: Option[URL
-      }
-      .mkString("\n")
-    val tmpFile: Path = iosp.mkTempFile()
-    iosp.writeFile(tmpFile, tblRepr)
+            writer.write(row)
+          case other =>
+            throw new EvalException(s"${other} should be an array", text, docSourceUrl)
+        }
+    } finally {
+      writer.close()
+    }
     V_File(tmpFile.toString)
   }
 
   // File write_map(Map[String, String])
   //
-  private def write_map(args: Vector[V], text: TextSource): V_File = {
+  // since: draft-1
+  protected def write_map(args: Vector[V], text: TextSource): V_File = {
     assert(args.size == 1)
     val m: V_Map =
       coercion.coerceTo(T_Map(T_String, T_String), args.head, text).asInstanceOf[V_Map]
-    val mapStrRepr = m.value
-      .map {
-        case (V_String(key), V_String(value)) => key + "\t" + value
-        case (k, v) =>
-          throw new EvalException(s"${k} ${v} should both be strings", text, docSourceUrl)
-      }
-      .mkString("\n")
     val tmpFile: Path = iosp.mkTempFile()
-    iosp.writeFile(tmpFile, mapStrRepr)
+    val writer = tmpFile.asCsvWriter[(String, String)](tsvConf)
+    try {
+      m.value
+        .foreach {
+          case (V_String(key), V_String(value)) => writer.write((key, value))
+          case (k, v) =>
+            throw new EvalException(s"${k} ${v} should both be strings", text, docSourceUrl)
+        }
+    } finally {
+      writer.close()
+    }
     V_File(tmpFile.toString)
   }
 
-  private def lineFromObject(obj: V_Object, text: TextSource): String = {
-    obj.members.values
-      .map { vw =>
-        coercion.coerceTo(T_String, vw, text).asInstanceOf[V_String].value
-      }
-      .mkString("\t")
+  private def lineFromObject(obj: V_Object, text: TextSource): Vector[String] = {
+    obj.members.values.map { vw =>
+      coercion.coerceTo(T_String, vw, text).asInstanceOf[V_String].value
+    }.toVector
   }
 
   // File write_object(Object)
   //
-  private def write_object(args: Vector[V], text: TextSource): V_File = {
+  // since: draft-1
+  // deprecation: removed in Version 2
+  protected def write_object(args: Vector[V], text: TextSource): V_File = {
     assert(args.size == 1)
     val obj = coercion.coerceTo(T_Object, args.head, text).asInstanceOf[V_Object]
-    val keyLine = obj.members.keys.mkString("\t")
-    val valueLine = lineFromObject(obj, text)
-    val content = keyLine + "\n" + valueLine
     val tmpFile: Path = iosp.mkTempFile()
-    iosp.writeFile(tmpFile, content)
+    val writer = tmpFile.asCsvWriter[Vector[String]](tsvConf)
+    try {
+      writer.write(obj.members.keys.toVector)
+      writer.write(lineFromObject(obj, text))
+    } finally {
+      writer.close()
+    }
     V_File(tmpFile.toString)
   }
 
   // File write_objects(Array[Object])
   //
-  private def write_objects(args: Vector[V], text: TextSource): V_File = {
+  // since: draft-1
+  // deprecation: removed in Version 2
+  protected def write_objects(args: Vector[V], text: TextSource): V_File = {
```

Comment:
It is good that this has been removed from version-2. It isn't well specified in the language.